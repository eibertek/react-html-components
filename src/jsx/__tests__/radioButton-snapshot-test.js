import React from 'react';
import RadioButton from '../radioButton.jsx';
import renderer from 'react/lib/ReactTestRenderer';

const radioButtonWithListeners = (
    <RadioButton required={true}
           label="test label"
           name="test_input"
           id="test"
           placeholder="RadioButton"
           extraClass="anotherClass"
           validate={true}
           errorMessage="wrong"
           successMessage="correct"
           changeCallback={function(){}}
           mouseEnterCallback={function(){}}
           mouseLeaveCallback={function(){}}/>
);

const radioButtonWithoutListeners = (
    <RadioButton required={true}
                 label="test label"
                 name="test_input"
                 checked={true}
                 id="test"/>
);

const radioButtonWithValue = (
    <RadioButton label="test label"
           name="test_input"
           id="test"
                 withGap={true}
                 checked={false}
           value="value"/>
);

describe('RadioButton', () => {
    it('with listeners matches snapshot', () => {
        const component = renderer.create(radioButtonWithListeners);

        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
    it('without listeners matches snapshot', () => {
        const component = renderer.create(radioButtonWithoutListeners);

        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
    it('with value matches snapshot', () => {
        const component = renderer.create(radioButtonWithValue);

        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});
