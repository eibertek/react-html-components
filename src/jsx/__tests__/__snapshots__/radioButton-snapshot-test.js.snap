exports[`RadioButton with listeners matches snapshot 1`] = `
<p>
  <input
    className="validate anotherClass"
    disabled={false}
    id="test"
    name="test_input"
    onChange={[Function bound _onChange]}
    onMouseEnter={[Function bound _onMouseEnter]}
    onMouseLeave={[Function bound _onMouseLeave]}
    placeholder="RadioButton"
    required={true}
    type="radio" />
  <label
    dangerouslySetInnerHTML={
      Object {
        "__html": "test label"
      }
    }
    data-error="wrong"
    data-success="correct"
    htmlFor="test" />
</p>
`;

exports[`RadioButton with value matches snapshot 1`] = `
<p>
  <input
    className=""
    disabled={false}
    id="test"
    name="test_input"
    onChange={[Function bound _onChange]}
    onMouseEnter={[Function bound _onMouseEnter]}
    onMouseLeave={[Function bound _onMouseLeave]}
    placeholder=""
    required={false}
    type="radio" />
  <label
    dangerouslySetInnerHTML={
      Object {
        "__html": "test label"
      }
    }
    data-error=""
    data-success=""
    htmlFor="test" />
</p>
`;

exports[`RadioButton without listeners matches snapshot 1`] = `
<p>
  <input
    className=""
    disabled={false}
    id="test"
    name="test_input"
    onChange={[Function bound _onChange]}
    onMouseEnter={[Function bound _onMouseEnter]}
    onMouseLeave={[Function bound _onMouseLeave]}
    placeholder=""
    required={true}
    type="radio" />
  <label
    dangerouslySetInnerHTML={
      Object {
        "__html": "test label"
      }
    }
    data-error=""
    data-success=""
    htmlFor="test" />
</p>
`;
